### ISSUE TITLE
[BUG] Norman V3 Cloud Credential Schema omits AWS config even though EKS driver is enabled

### ISSUE THREAD

**Issue by nwmac** [rancher/rancher#39399]
This issues described the root causes of this UI issue: https://github.com/rancher/dashboard/issues/5962

The API `/v3/schema/cloudcredential` returns the schema for the cloud credential resource. This data includes the `resourceFields` property that includes the fields for the credentials for the enabled cloud providers  - e.g.

![image](https://user-images.githubusercontent.com/1955897/197756110-0e541679-45ad-446b-a764-c20327ed21e2.png)

When the 'Amazon EC2' node driver is enabled, this API correctly includes the `amazonec2credentialConfig` field in `resourceFields`.

If, however, you disable the EC2 node driver but enable the 'Amazon EKS` cluster driver, the field is omitted for the AWS credential, where it should be included - this is the bug that needs fixing.

In other words, the AWS Credential only shows up when the EC2 Node driver is enabled and it ignores the EKS cluster driver status.

Note, this works correctly for other credentials (e.g. google) that are used by both GKE and Google - here the `googlecredentialConfig` field shows correctly if either the Google Cluster OR Node driver are active.

**Comment by HarrisonWAffel** [comment]
I've created a PR to update the behavior for both Amazon EC2 as well as Azure. Now, disabling the node driver for either of those providers will not result in their resource field being removed from the api schema. This matches the behavior of the existing google cloud credential resource field.

**Comment by jameson-mcghee** [comment]
When attempting to reproduce this issue on `v2.6.7`, I was able to see the `resourceFields` `amazonec2credentialconfig` and `azurecredentialconfig` removed from the `/v3/schema/cloudcredential` config as described when following the Steps to Reproduce. However, then attempting to reproduce this issue on `v2.7-head` (Commit ID: 38979f8), the issue was no longer reproducing. Therefore I am closing this ticket.

**Steps To Reproduce:**

1. Create a new rancher server instance
2. Navigate to `https://<CLUSTER_URL>/v3/schema/cloudcredential` and ensure that the `resourceFields` `amazonec2credentialconfig` and `azurecredentialconfig` are present
3. Return to the Rancher UI and navigate to the Cluster Management --> Drivers page
4. Under 'Cluster Drivers' Deactivate the `Amazon EKS` and `Azure AKS` Cluster Drivers
5. Under 'Node Drivers' Deactivate the `Amazon EC2` and `Azure` Node Drivers
6. Navigate back to `https://<CLUSTER_URL>/v3/schema/cloudcredential` and ensure that the `resourceFields` `amazonec2credentialconfig` and `azurecredentialconfig` are still present, even though both of the node drivers are disabled.

**`v2.6.7` (Before Driver Deactivation):**
![image](https://user-images.githubusercontent.com/13294321/205399887-b398ea74-4ae9-4aff-98d2-9a5a9c1ba2e9.png)
![image](https://user-images.githubusercontent.com/13294321/205399903-716d0c97-3e69-4ec9-a15d-b6593c46d4ae.png)
![image](https://user-images.githubusercontent.com/13294321/205399912-85e9ef02-6900-4cc6-bf2d-4992d214cbe0.png)

**`v2.6.7` (After Driver Deactivation):**
![image](https://user-images.githubusercontent.com/13294321/205399985-93242ee3-7db5-46f8-ba00-93df7af10a5d.png)
![image](https://user-images.githubusercontent.com/13294321/205399997-33027051-1e73-46ed-9b76-b096ebd6e9ea.png)
![image](https://user-images.githubusercontent.com/13294321/205400012-a65db0ee-03bf-40d6-b2e8-81a5ee21fca4.png)

**`v2.7-head` (Commit ID: 38979f8) (Before Driver Deactivation):**
![image](https://user-images.githubusercontent.com/13294321/205400082-aa9c3238-e8b6-4a9d-8dbb-82dd5cfb0ceb.png)
![image](https://user-images.githubusercontent.com/13294321/205400092-e3b330ff-3a25-41d6-984c-6d64e1dd4eaa.png)
![image](https://user-images.githubusercontent.com/13294321/205400094-52bfb35d-1689-42a2-8a9c-1ae6b64f8824.png)

**`v2.7-head` (Commit ID: 38979f8) (After Driver Deactivation):**
![image](https://user-images.githubusercontent.com/13294321/205400109-afa860c7-45c9-402f-80e5-483cb8ffded1.png)
![image](https://user-images.githubusercontent.com/13294321/205400116-d145031c-7459-4705-b4fb-62ce02294699.png)
![image](https://user-images.githubusercontent.com/13294321/205400120-95261bc1-6ab8-4776-810c-7f556ce4bb08.png)
